scapy/contrib/pim.py

```diff
@@ -238,6 +274,73 @@ class PIMv2JoinPrune(_PIMGenericTlvBase):
     ]
 
 
+
+#########################################################
+# IPv6 
+class PIMv2JoinPruneAddr6sBase(_PIMGenericTlvBase):
+    name = "PIMv2 Join: IPv6 Address Base"
+    fields_desc = [
+        ByteField("addr_family", 2),
+        ByteField("encoding_type", 0),
+        BitField("rsrvd", 0, 5),
+        BitField("sparse", 0, 1),
+        BitField("wildcard", 0, 1),
+        BitField("rpt", 1, 1),
+        ByteField("mask_len", 128),
+        IP6Field("src_ip", "::0")
+
+    ]
+
+class PIMv2JoinAddr6s(PIMv2JoinPruneAddr6sBase):
+    name = "PIMv2 Join: Source IPv6 Address"
+
+
+class PIMv2PruneAddr6s(PIMv2JoinPruneAddr6sBase):
+    name = "PIMv2 Prune: Source IPv6 Address"
+
+
+class PIMv2GroupAddr6s(_PIMGenericTlvBase):
+    name = "PIMv2 Join/Prune: Multicast Group Address IPv6"
+    fields_desc = [
+        ByteField("addr_family", 2),
+        ByteField("encoding_type", 0),
+        BitField("bidirection", 0, 1),
+        BitField("reserved", 0, 6),
+        BitField("admin_scope_zone", 0, 1),
+        ByteField("mask_len", 128),
+        IP6Field("gaddr", "::0"),
+        BitFieldLenField("num_joins", None, size=16, count_of="join_ips"),
+        BitFieldLenField("num_prunes", None, size=16, count_of="prune_ips"),
+        PacketListField("join_ips", [], PIMv2JoinAddr6s,
+                        count_from=lambda x: x.num_joins),
+        PacketListField("prune_ips", [], PIMv2PruneAddr6s,
+                        count_from=lambda x: x.num_prunes),
+    ]
+
+class PIMv2JoinPruneIpv6(_PIMGenericTlvBase):
+    name = "PIMv2 Join/Prune Options for IPv6"
+    fields_desc = [
+        ByteField("up_addr_family", 2),
+        ByteField("up_encoding_type", 0),
+        IP6Field("up_neighbor_ip", "::0"),
+        ByteField("reserved", 0),
+        FieldLenField("num_group", None, count_of="jp_ips", fmt="B"),
+        ShortField("holdtime", 210),
+        PacketListField("jp_ips", [], PIMv2GroupAddr6s,
+                        count_from=lambda pkt: pkt.num_group)
+    ]
+
+class PIMv2JoinPrune(Packet):
```

Comment:
I think you could keep a single Prune class thanks to using a `MultipleTypeField`. It would probably look a bit better